src/services/action/repositories/action.ts

```diff
@@ -167,12 +191,25 @@ export class ActionRepository {
       subquery.andWhere('action.type IN (:...types)');
     }
 
+    if (filters?.startDate) {
+      subquery.andWhere('action.created_at >= :startDate', {
+        startDate: filters.startDate,
+      });
+    }
+
+    if (filters?.endDate) {
+      subquery.andWhere('action.created_at <= :endDate', {
+        endDate: filters.endDate,
+      });
+    }
```

Comment:
I think we should check the validity of the range, eg startDate cannot be "higher" than `endDate`. Let's throw an error if it's not valid. And let's do it for both endpoints.